store/src/java-test/com/zimbra/cs/filter/GetFilterRulesTest.java

```diff
@@ -0,0 +1,427 @@
+                MockProvisioning.DEFAULT_ACCOUNT_ID);
+        RuleManager.clearCachedRules(account);
+        account.setMailSieveScript(filterScript);
+
+        Element request = new Element.XMLElement(MailConstants.GET_FILTER_RULES_REQUEST);
+        Element response = new GetFilterRules().handle(request, ServiceTestUtil.getRequestContext(account));
+
+        String expectedSoapResponse =
+              "<GetFilterRulesResponse xmlns=\"urn:zimbraMail\">"
+              + "<filterRules>"
+                + "<filterRule active=\"1\">"
+                  + "<filterTests>"
+                    + "<headerTest stringComparison=\"matches\" header=\"Subject\" index=\"0\" value=\"*\"/>"
+                  + "</filterTests>"
+                  + "<nestedRule>"
+                    + "<filterTests condition=\"anyof\">"
+                      + "<headerTest stringComparison=\"matches\" header=\"From\" index=\"0\" value=\"*\"/>"
+                      + "<headerTest stringComparison=\"matches\" header=\"To\"   index=\"1\" value=\"*\"/>"
+                    + "</filterTests>"
+                    + "<filterActions>"
+                      + "<actionFileInto folderPath=\"nested-if-block\" index=\"0\" copy=\"0\"/>"
+                    + "</filterActions>"
+                  + "</nestedRule>"
+                + "</filterRule>"
+              + "</filterRules>"
+            + "</GetFilterRulesResponse>";
+        XMLDiffChecker.assertXMLEquals(expectedSoapResponse, response.prettyPrint());
+    }
+
+    @Test
+    public void testWithoutIfBlock1() throws Exception {
+        // - no if block
+        String filterScript
+                    = "require \"tag\";"
+                    + "fileinto \"no-if-block\";";
+        Account account = Provisioning.getInstance().getAccount(
+                MockProvisioning.DEFAULT_ACCOUNT_ID);
+        RuleManager.clearCachedRules(account);
+        account.setMailSieveScript(filterScript);
+
+        Element request = new Element.XMLElement(MailConstants.GET_FILTER_RULES_REQUEST);
+        Element response = new GetFilterRules().handle(request, ServiceTestUtil.getRequestContext(account));
+
+        String expectedSoapResponse =
+                "<GetFilterRulesResponse xmlns=\"urn:zimbraMail\">"
+                + "<filterRules>"
+                  + "<filterRule active=\"1\">"
+                    + "<filterTests>"
+                      + "<trueTest index=\"0\"/>"
+                    + "</filterTests>"
+                    + "<filterActions>"
+                      + "<actionFileInto folderPath=\"no-if-block\" index=\"0\" copy=\"0\"/>"
+                    + "</filterActions>"
+                  + "</filterRule>"
+                + "</filterRules>"
+              + "</GetFilterRulesResponse>";
+        XMLDiffChecker.assertXMLEquals(expectedSoapResponse, response.prettyPrint());
+    }
+
+    @Test
+    public void testWithoutIfBlock2() throws Exception {
```

Comment:
Each test case does call the assertFalse() inside the XMLDiffChecker.assertXMLEquals().